src/repositories/live.repository.ts

```diff
@@ -0,0 +1,330 @@
+      method: 'POST',
+      form: {
+        _csrftoken: this.client.state.cookieCsrfToken,
+        _uuid: this.client.state.uuid,
+      },
+    });
+    return body;
+  }
+
+  public async getQuestions(): Promise<LiveGetQuestionsResponseRootObject> {
+    const { body } = await this.client.request.send<LiveGetQuestionsResponseRootObject>({
+      url: '/api/v1/live/get_questions/',
+      method: 'GET',
+    });
+    return body;
+  }
+
+  public async wave(broadcastId: string, viewerId: string): Promise<LiveWaveResponseRootObject> {
+    const { body } = await this.client.request.send<LiveWaveResponseRootObject>({
+      url: `/api/v1/live/${broadcastId}/wave/`,
+      method: 'POST',
+      form: this.client.request.sign({
+        viewer_id: viewerId,
+        _csrftoken: this.client.state.cookieCsrfToken,
+        _uid: this.client.state.cookieUserId,
+        _uuid: this.client.state.uuid,
+      }),
+    });
+    return body;
+  }
+
+  public async like(broadcastId: string, likeCount: number = 1): Promise<LiveLikeResponseRootObject> {
+    const { body } = await this.client.request.send<LiveLikeResponseRootObject>({
+      url: `/api/v1/live/${broadcastId}/like/`,
+      method: 'POST',
+      form: this.client.request.sign({
+        _csrftoken: this.client.state.cookieCsrfToken,
+        _uid: this.client.state.cookieUserId,
+        _uuid: this.client.state.uuid,
+        user_like_count: likeCount,
+      }),
+    });
+    return body;
+  }
+
+  public async getLikeCount(broadcastId: string, likeTs: string = ''): Promise<LiveLikeCountResponseRootObject> {
+    const { body } = await this.client.request.send<LiveLikeCountResponseRootObject>({
+      url: `/api/v1/live/${broadcastId}/get_like_count/` + (likeTs === '' ? '' : `?like_ts=${likeTs}`),
+      method: 'GET',
+    });
+    return body;
+  }
+
+  public async resumeBroadcastAfterContentMatch(broadcastId: string): Promise<any> {
+    // TODO: test
+    const { body } = await this.client.request.send({
+      url: `/api/v1/live/${broadcastId}/resume_broadcast_after_content_match/`,
+      method: 'POST',
+      form: this.client.request.sign({
+        _csrftoken: this.client.state.cookieCsrfToken,
+        _uid: this.client.state.cookieUserId,
+        _uuid: this.client.state.uuid,
+      }),
+    });
+    return body;
+  }
+
+  public async getJoinRequestCounts(
+    broadcastId: string,
+    lastTotalCount: number = 0,
+    lastSeenTs: number = 0,
+    lastFetchTs: number = 0)
+    : Promise<LiveJoinRequestCountsResponseRootObject> {
+    const baseUrl = `/api/v1/live/${broadcastId}/get_join_request_counts/`;
+    const { body } = await this.client.request.send<LiveJoinRequestCountsResponseRootObject>({
+      url: `${baseUrl}?last_total_count${lastTotalCount}&last_seen_ts=${lastSeenTs}&last_seen_ts${lastFetchTs}`,
+      method: 'GET',
+    });
+    return body;
+  }
+
+  public getObsSettings(info: LiveCreateBroadcastResponseRootObject): LiveObsSettings {
```

Comment:
This method should be static, it will be better to place it in `entities` as static method. And will be nice to rename it with more human-friendly name.

Reply:
One more thing, you shouldn't expect whole `LiveCreateBroadcastResponseRootObject` cause you don't need it.
It'll be better to describe the signature like
```ts
public getObsSettings(info: {upload_url: string, broadcast_id: string})
```

Reply:
I don't exactly understand what this function does, so can't suggest the most suitable entity.

Reply:
It fits in the entities. This is intended for OBS (or any other streaming software) as it expects a key and url.